src/test/java/uk/nhs/ctp/service/ValidationServiceTest.java

```diff
@@ -51,4 +58,164 @@ public void zip_creation() throws IOException {
+        .requestUrl("http://valid.com/request/url1")
+        .dateOfEntry(now)
+        .build();
+    var entry2 = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("<body>validRequestBody2</body>")
+        .requestUrl("http://valid.com/request/url2")
+        .dateOfEntry(then)
+        .build();
+    var entry3 = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("{ \"body\": \"validRequestBody3\" }")
+        .requestUrl("http://valid.com/request/url3")
+        .dateOfEntry(then)
+        .build();
+
+    var audit1 = AuditSession.builder()
+        .entry(entry1)
+        .entry(entry2)
+        .additionalProperty("caseId", "11")
+        .build();
+    var audit2 = AuditSession.builder()
+        .entry(entry3)
+        .additionalProperty("caseId", "22")
+        .build();
+
+    var audits = List.of(audit1, audit2);
+
+    var zipped = validationService.zipAudits(audits, OperationType.ENCOUNTER);
+
+    assertThat(unzipEntries(zipped), containsInAnyOrder(
+        matchesAuditEntry("encounter11/valid.com/request/url1.json", entry1),
+        matchesAuditEntry("encounter11/valid.com/request/url2.xml", entry2),
+        matchesAuditEntry("encounter22/valid.com/request/url3.json", entry3)
+    ));
+  }
+
+  @Test
+  public void zipAudits_withServiceSearchAudits_shouldZipEntryData() throws IOException {
+    var entry = AuditEntry.builder()
+        .requestMethod("GET")
+        .responseBody("{ \"body\": \"validRequestBody1\" }")
+        .requestUrl("http://valid.com/request/url1")
+        .dateOfEntry(Instant.now())
+        .build();
+
+    var audits = singletonList(AuditSession.builder().entry(entry).build());
+
+    var zipped = validationService.zipAudits(audits, OperationType.SERVICE_SEARCH);
+
+    assertThat(unzipEntries(zipped), contains(
+        matchesAuditEntry("service_search/valid.com/request/url1.json", entry)
+    ));
+  }
+
+  private List<ZippedAuditEntry> unzipEntries(byte[] bytes) throws IOException {
+    try (var input = new ByteArrayInputStream(bytes)) {
+      try (var zip = new ZipInputStream(input)) {
+        var entries = new ArrayList<ZippedAuditEntry>();
+
+        ZipEntry zipEntry;
+        while ((zipEntry = zip.getNextEntry()) != null) {
+          entries.add(ZippedAuditEntry.builder()
+              .instant(zipEntry.getCreationTime().toInstant())
+              .path(zipEntry.getName())
+              .body(new String(zip.readAllBytes(), UTF_8))
+              .build());
+        }
+        zip.closeEntry();
+
+        return entries;
+      }
+    }
+  }
+
+  @Value
+  @Builder
+  private static class ZippedAuditEntry {
+    String path;
+    String body;
+    Instant instant;
+  }
+
+  private Matcher<ZippedAuditEntry> matchesAuditEntry(String path, AuditEntry auditEntry) {
+    return new FunctionMatcher<>(
+        zipEntry -> path.equals(zipEntry.getPath())
+            && sameSecond(auditEntry.getDateOfEntry(), zipEntry.getInstant())
```

Comment:
if we're doing date assertions we should really be mocking the instant so that we can check is is _exactly_ the same instant.
```suggestion
            && auditEntry.getDateOfEntry().equals(zipEntry.getInstant())
```

Reply:
Tried that initially but the `Instants` would not be precisely equal as the zipping process truncates the `Instant` precision to the second, removing the relevant milliseconds from `zipEntry`.

Reply:
In that case you'd need to create a new instant without the milliseconds on and expect that :)